src/Workspaces/Core/Portable/Utilities/TaskQueue.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Roslyn.Utilities
+{
+    internal sealed class TaskQueue
+    {
+        private readonly IAsynchronousOperationListener _operationListener;
+        private readonly TaskScheduler _taskScheduler;
+
+        private readonly object _gate = new object();
+        private Task _latestTask;
+
+        internal TaskQueue(IAsynchronousOperationListener operationListener, TaskScheduler taskScheduler)
```

Comment:
```suggestion
        public TaskQueue(IAsynchronousOperationListener operationListener, TaskScheduler taskScheduler)
```

IDE style (and common in .net) is to keep the members public if the final accessibility is still constrained by the type.  it's basically saying "i expect this to be usable by anyone who can access this".  "internal" is more generally used to mean "actually, only a subset of the world that *can* access this *should* access this".